tests/pool.ts

```diff
@@ -1,11 +1,23 @@
 import {
+  assert,
   assertEquals,
   assertThrowsAsync,
 } from "../test_deps.ts";
 import { Pool } from "../pool.ts";
 import { delay } from "../utils.ts";
 import { TEST_CONNECTION_PARAMS, DEFAULT_SETUP } from "./constants.ts";
 
+Deno.test("string pool connection", async function () {
+  const { user, password, database, hostname, port } = TEST_CONNECTION_PARAMS;
+  const pool = new Pool(
+    `postgres://${user}:${password}@${hostname}:${port}/${database}`,
+    10,
+  );
+  const result = await pool.query("SELECT * FROM ids;");
```

Comment:
Do you need to await for it to be connected?

..I didn't think you did (feels like there is a bug for sure :) )

What is testPool doing different to this? Since those tests using testPool don't seem to suffer from it? (Is it the setupQueries somehow?)

Reply:
I don't know. Test Pool looks to just run some setup queries, but we don't need them. I can change the SQL to be `SELECT true;` to remove a layer of potential errors; but the error we're getting back looks cryptic.

Reply:
Yep, definitely feels like a bug. My bet is something to do with fewer await and not being lazy is what's exposing it.

Reply:
I've added `await x.connect()`. Now getting failures of resource leakage! Not sure where its leaking, calling `await x.end()` so it should be fine, right?